src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1406,13 +1378,47 @@ int fx_muxer_t::execute(
     return result;
 }
 
-int fx_muxer_t::get_com_activation_delegate(
-    const host_startup_info_t &host_info,
-    void **delegate)
+
+static int get_delegate_from_runtime(
+    const pal::string_t& impl_dll_dir,
+    corehost_init_t* init,
+    coreclr_delegate_type type,
+    void** delegate)
 {
-    assert(host_info.is_valid());
+    pal::dll_t corehost;
+    hostpolicy_contract host_contract{};
+    corehost_get_delegate_fn coreclr_delegate = nullptr;
+
+    int code = load_hostpolicy(impl_dll_dir, &corehost, host_contract, "corehost_get_coreclr_delegate", &coreclr_delegate);
+    if (code != StatusCode::Success)
```

Comment:
We should write out a good error message in this case. This can happen if somebody is trying to run COM or IJW on .NET Core 2.* since the `hostpolicy` of 2.* won't have the necessary export. We should be able to detect that specific case and report an error in lines of "The COM component must target netcoreapp3.0 or higher version" or something like that.